source/MessageHandlers/SendDataHandler.ts

```diff
@@ -108,20 +121,23 @@ export default class SendDataHandler implements IMessageHandler {
                             await reader.close();
                         }
                     }
-                    queryCallback(null, { name: queryStatement.name, data: itemMetadata});
+                    queryCallback(null, { name: queryDefinition.name, data: itemMetadata});
                 },
                 async (err, results) => { // async.mapLimit callback
                     if (err) {
                         await this._connectionPool.close();
                         reject(err);
                     } else {
+                        await this.ingestDDL(validTables, folderPath);
```

Comment:
Ingest the DDL records before closing the pool. The file should be exactly the same as prior 'manifestations' of the DDL fetching logic.